app/controllers/actions/IdentifierAction.scala

```diff
@@ -65,18 +65,25 @@ class AuthenticatedIdentifierAction @Inject() (
               checkUserAllowList(enrolment.value)(hc).flatMap { _ =>
                 block(IdentifierRequest(request, internalId, enrolment.value, affinityGroup))
               }
+            case Some(enrolment) if enrolment.value.isBlank  =>
+              throw InternalError("EORI is empty")
```

Comment:
Which recovery case does this end up in?

Reply:
It falls into the AuthorisationException, perhaps I should try using that instead of InternalError

Reply:
Upon looking @ChrisCookOC this is the correct one to use: `
object AuthorisationException {

  def fromString(reason: String): AuthorisationException = reason match {
    case "InsufficientConfidenceLevel" => new InsufficientConfidenceLevel
    case "UnsupportedAffinityGroup"    => new UnsupportedAffinityGroup
    case "UnsupportedCredentialRole"   => new UnsupportedCredentialRole
    case "UnsupportedAuthProvider"     => new UnsupportedAuthProvider
    case "BearerTokenExpired"          => new BearerTokenExpired
    case "MissingBearerToken"          => new MissingBearerToken
    case "InvalidBearerToken"          => new InvalidBearerToken
    case "SessionRecordNotFound"       => new SessionRecordNotFound
    case "IncorrectCredentialStrength" => new IncorrectCredentialStrength
    case "InsufficientEnrolments"      => new InsufficientEnrolments
    case "FailedRelationship"          => new FailedRelationship
    case "IncorrectNino"               => IncorrectNino
    case other                         => InternalError(other)
  }
}`

Reply:
Ready for review